### PR TITLE
test: move sync_* implementations to framework member functions

### LINKED ISSUE #18930: test: Remove sync_blocks global
sync_blocks has been made a member of the test framework in https://github.com/bitcoin/bitcoin/pull/15773, but the global helper has been left intact for now.

It has been suggested to move the implementation to the member function as well https://github.com/bitcoin/bitcoin/pull/15773#pullrequestreview-224235071

Implementing the functionality in a single place in the test framework allows the method to take account of contextual information, such as global timeout modifications. Also, having a global that does almost-but-not-quite the same thing is confusing for test writers and reviewers.

So the implementation should be moved, along with removing the global helper.

#### Useful skills:

Python3, basic understanding of the test framework

#### Want to work on this issue?

The purpose of the `good first issue` label is to highlight which issues are suitable for a new contributor without a deep understanding of the codebase.

You do not need to request permission to start working on this. You are encouraged to comment on the issue if you are planning to work on it. This will help other contributors monitor which issues are actively being addressed and is also an effective way to request assistance if and when you need it.

For guidance on contributing, please read [CONTRIBUTING.md](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md) before opening your pull request.

### PR DESCRIPTION
**Pull request by brakmic**
This PR moves global `util.py` implementations of **sync_blocks** and **sync_mempools** to `framework/test_framework.py` member functions.

Additionally, it updates `feature_backwards_compatibility.py`-test to use these new member functions.

Fixes https://github.com/bitcoin/bitcoin/issues/18930

### CODE REVIEW

File: test/functional/feature_backwards_compatibility.py [hunk 1]
```diff
@@ -27,9 +27,7 @@
 
 from test_framework.util import (
     adjust_bitcoin_conf_for_pre_17,
-    assert_equal,
-    sync_blocks,
-    sync_mempools,
+    assert_equal
```
Comment (MarcoFalke) [inline test/functional/feature_backwards_compatibility.py:30]: nit: Please keep the trailing comma to minimize this diff and future diffs, as well as guiding editors with formatting.

File: test/functional/test_framework/test_framework.py [hunk 2]
```diff
@@ -30,9 +30,7 @@
     connect_nodes,
     disconnect_nodes,
     get_datadir_path,
-    initialize_datadir,
-    sync_blocks,
-    sync_mempools,
+    initialize_datadir
```
Comment (MarcoFalke) [inline test/functional/test_framework/test_framework.py:33]: same

**Review by maflcko — APPROVED** [review]
Approach ACK (only looked at the diff on GitHub)

File: test/functional/test_framework/test_framework.py [hunk 3]
```diff
@@ -519,11 +517,49 @@ def join_network(self):
...
+        """
+        Wait until everybody has the same tip.
 
-    def sync_mempools(self, nodes=None, **kwargs):
-        sync_mempools(nodes or self.nodes, **kwargs)
+        sync_blocks needs to be called with an rpc_connections set that has least
+        one node already synced to the latest, stable tip, otherwise there's a
+        chance it might return before all nodes are stably synced.
+        """
+        stop_time = time.time() + timeout
```
Comment (MarcoFalke) [inline test/functional/test_framework/test_framework.py:529]: The timeout should be scaled by `self.factor`, no? If yes, please add a commit for that.

Comment (brakmic) [reply test/functional/test_framework/test_framework.py]: Wasn't aware of it as the methods weren't adapted in the *factor*-PR. Have used the original `self.options.factor` property to adjust timeouts for both methods.

**Comment by MarcoFalke** [comment]
I liked the solution. Looking forward to seeing this reopened :)

**Comment by practicalswift** [comment]
Me too -- please consider re-opening :)

**Pull request closed without merging**